scripts/gen_eval_files.lua

```diff
@@ -385,6 +391,46 @@ local function render_eval_meta(f, fun, write)
   write(render_fun_sig(funname, params))
 end
 
+--- @return table<string, vim.ExCmd>
+local function get_excmd_meta()
+  --- @type vim.ExCmd[]
+  local cmds = require('src/nvim/ex_cmds').cmds
+  local cmd_table = {} ---@type table<string, vim.ExCmd>
+  for _, value in ipairs(cmds) do
+    cmd_table[value.command] = value
+  end
+  return cmd_table
+end
+
+--- @param cmd_name string
+--- @param cmd vim.ExCmd
+--- @param write fun(line: string)
+local function render_excmd_meta(cmd_name, cmd, write)
+  assert(cmd_name == cmd.command)
+  if cmd.removed then
+    -- skip deprecated or removed (E319) commands
+    return
+  end
+
+  local function write_vimcmd_fun(name, full_name)
```

Comment:
Ideally we should also generate the vimdoc files and add the description into `ex_cmds.lua` but that can be a future improvement.

Basically everything we did for `vim.fn.*`

Reply:
Exactly, we should do so. I broke down the scope/step because `vim.fn.*` was easier as all the builtin vimscript functions are listed in `docbuiltin.txt` in the alphabetical order, but all the Ex cmds are scattered with different format.

Reply:
Sure I understand that. This is a big task.